### PR TITLE
Node library: invalid type of default property in _indexConfig #9856

### PR DESCRIPTION
**Pull request by ComLock**

### CODE REVIEW

**Review by tajakobsen — COMMENTED** [review]
Can you confirm that `configs` isn't optional.

File: modules/lib/lib-node/src/main/resources/lib/xp/node.ts [hunk 1]
```diff
@@ -762,14 +762,30 @@ export interface AccessControlEntry {
...
+    | 'none'
+    | 'byType'
+    | 'fulltext'
+    | 'path'
+    | 'minimal';
+
+export interface NodeIndexConfigParams {
+    analyzer?: string;
+    default?: Partial<NodeConfigEntry> | NodeIndexConfigTemplates;
+    configs: {
```
[the remark this answers is not in the document]
Comment (ComLock) [reply modules/lib/lib-node/src/main/resources/lib/xp/node.ts:783]: I'm not sure, maybe we have some tests that cover this? @edloidas

[the remark this answers is not in the document]
Comment (ComLock) [reply modules/lib/lib-node/src/main/resources/lib/xp/node.ts:783]: Ah configs, probably optional

[the remark this answers is not in the document]
Comment (ComLock) [reply modules/lib/lib-node/src/main/resources/lib/xp/node.ts:783]: My mind was on the properties of NodeConfigEntry

[the remark this answers is not in the document]
Comment (ComLock) [reply modules/lib/lib-node/src/main/resources/lib/xp/node.ts:783]: I think the properties of NodeConfigEntry are optional, but I'm not certain.

**Review by tajakobsen — APPROVED** [review]
Looks good

**Review by edloidas — APPROVED** [review]

**Review by rymsha — APPROVED** [review]

**Pull request merged**